scripts/scil_connectivity_math.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import argparse
+import logging
+import os
+
+from dipy.io.utils import is_header_compatible
+import nibabel as nib
+import numpy as np
+
+from scilpy.image.operations import (is_float, absolute_value, around,
+                                     addition, ceil, convert,
+                                     difference, division, floor, intersection,
+                                     invert, lower_clip, lower_threshold, mean,
+                                     multiplication, normalize_max,
+                                     normalize_sum, std, subtraction,
+                                     union, upper_threshold, upper_clip,
+                                     get_array_operation_doc, 
+                                     get_image_operation_doc)
+from scilpy.io.utils import (add_overwrite_arg,
+                             add_verbose_arg,
+                             assert_outputs_exist)
+
+OPERATIONS = {
+    'lower_threshold': lower_threshold,
+    'upper_threshold': upper_threshold,
+    'lower_clip': lower_clip,
+    'upper_clip': upper_clip,
+    'absolute_value': absolute_value,
+    'round': around,
+    'ceil': ceil,
+    'floor': floor,
+    'normalize_sum': normalize_sum,
+    'normalize_max': normalize_max,
+    'convert': convert,
+    'invert': invert,
+    'addition': addition,
+    'subtraction': subtraction,
+    'multiplication': multiplication,
+    'division': division,
+    'mean': mean,
+    'std': std,
+    'union': union,
+    'intersection': intersection,
+    'difference': difference,
+}
+
+DESCRIPTION = """
+Performs an operation on a list of matrices. The supported operations are 
+listed below.
+
+Some operations such as multiplication or addition accept float value as
+parameters instead of matrices.
+> scil_connectivity_math.py multiplication mat.npy 10 mat mult_10.npy
+"""
+
+ADDED_DOC = get_array_operation_doc().replace('images', 'matrices')
+ADDED_DOC = ADDED_DOC.replace('image', 'matrix')
+ADDED_DOC = ADDED_DOC.replace('IMG', 'MAT')
+DESCRIPTION += ADDED_DOC
+
+
+def _build_args_parser():
+    p = argparse.ArgumentParser(
+        formatter_class=argparse.RawTextHelpFormatter,
+        description=DESCRIPTION)
+
+    p.add_argument('operation',
+                   help='The type of operation to be performed on the '
+                   'matrices.')
+
+    p.add_argument('inputs', nargs='+',
+                   help='The list of matrices files or parameters.')
+
+    p.add_argument('--data_type',
```

Comment:
Should we provide choices?

Reply:
Maybe already answer but should we ? Because I don't know what to give. f64, float64 or np.float64 ?